pkg/publisher/combo/fanout.go

```diff
@@ -102,13 +116,49 @@ func (f *fanoutPublisher) PublishShardResult(
 		return p.PublishShardResult(ctx, shard, hostID, shardResultPath)
 	})
 
-	// Just return the first storage spec that we get
-	select {
-	case value := <-valueChannel:
-		return value, nil
-	case err := <-errorChannel:
-		return model.StorageSpec{}, err
+	type result struct {
+		Sender publisher.Publisher
+		Value  model.StorageSpec
 	}
+
+	timeoutChannel := make(chan bool, 1)
+	results := make([]result, 0)
```

Comment:
A very minor nitpick is that this could probably be `make([]result, 0, len(publishers))` as we are expecting a result from every publsiher.

Reply:
Makes sense!